src/availability.rs

```diff
@@ -209,6 +215,33 @@ where
             .try_flatten_stream()
             .boxed()
         })?
+        .get("get_payload", |req, state| {
+            async move {
+                let id = if let Some(height) = req.opt_integer_param("height")? {
+                    BlockId::Number(height)
+                } else if let Some(hash) = req.opt_blob_param("hash")? {
+                    BlockId::PayloadHash(hash)
+                } else {
+                    BlockId::Hash(req.blob_param("block-hash")?)
+                };
+                state.get_payload(id).await.context(FetchBlockSnafu {
+                    resource: id.to_string(),
```

Comment:
Does it makes sense to include some of the `req` context in the error? I don't recall whether tide disco handles this

Reply:
We include the ID of the resource we failed to fetch (which is derived from `req`). Anything else you'd want to see?

Reply:
hmmm nothing in particular comes to mind!